src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react';
+import { getPeople } from '../api';
+import { Loader } from '../components/Loader';
+import { Person } from '../types';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { PeopleTable } from '../components/PeopleTable';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [hasLoadedOnce, setHasLoadedOnce] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(data => {
+        const updatedPeople = data.map(person => ({
```

Comment:
This is a performance issue. The 'find' method runs in O(n) time. Since you're using it inside a 'map' method, which also runs in O(n) time, the overall time complexity is O(n^2). This might cause performance problems if the 'data' array is large. Consider creating a dictionary of people by name first, and then use it to look up the parents. This will reduce the time complexity to O(n).